src/test/java/com/google/sps/ProfileTest.java

```diff
@@ -120,4 +137,38 @@ public void doGetReturnsCorrectResponseForTutor() throws Exception {
         String expected = new Gson().toJson(expectedTutor);
         Assert.assertTrue(stringWriter.toString().contains(expected));
     }
+
+    @Test
+    public void doPostCancelsFormSubmit() throws Exception {
+        when(request.getParameter("submit")).thenReturn("Cancel");
+
+        servlet.doPost(request, response);
+
+        ArgumentCaptor<String> url = ArgumentCaptor.forClass(String.class);
```

Comment:
What is an ArgumentCaptor used for?

Reply:
It captures argument values, in this case for the sendRedirect, so I can check that the servlet is redirecting where I want it to.